lib/course-db.js

```diff
@@ -78,7 +78,8 @@ module.exports.loadCourseInfo = function(courseInfo, courseDir, logger, callback
         });
 
         // Course options
-        courseInfo.options = {};
+        courseInfo.options = info.options || {};
+        // Set default course options
```

Comment:
I think that it's probably cleaner here to stick with @nwalters512's existing pattern of starting with `{}` and then explicitly copying over each option that we know about, with a default explicitly set.

Reply:
It seems like wasteful duplication to define a strict schema, and then duplicate that checking and validation with a different library on load. I get that adding missing default valuables is important, and missing in our json validator-- but if that existed, would we need to be so strict here?

Reply:
I agree that if we had really strict JSON schema checking then we wouldn't need to do it here. Until that time I'd rather be strict.